### PR TITLE
Empty samples page with reset button

### PR DESCRIPTION
**Pull request by manivoxel51**
Screen Shot 2023-03-08 at 3 55 09 PM
Screen Shot 2023-03-08 at 3 55 15 PM
## What changes are proposed in this pull request?

Shows an Empty component for when there are no samples in a dataset or there are no samples in the given view.

Screen Shot 2023-03-08 at 3 54 52 PM
Screen Shot 2023-03-08 at 3 54 57 PM

(Please fill in changes proposed in this fix)

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/core/src/components/EmptySamples.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+`;
+
+const ActionContainer = styled(TextContainer)`
+  justify-content: start;
+  display: flex;
+  position: relative;
+`;
+
+function EmptySamples() {
+  const loadedView = useRecoilValue<fos.State.Stage[]>(fos.view);
```
Comment (sashankaryal) [inline app/packages/core/src/components/EmptySamples.tsx:39]: nit: type parameter implicitly derived

File: app/packages/core/src/components/EmptySamples.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    fos.count({ path: "", extended: false, modal: false })
+  );
+  const setView = fos.useSetView();
+
+  const showEmptySamples = useMemo(
+    () => loadedView?.length && !totalSamples,
+    [totalSamples, loadedView]
+  );
+
+  if (!showEmptySamples) return null;
```
Comment (sashankaryal) [inline app/packages/core/src/components/EmptySamples.tsx:50]: nit: newline before return

**Review by sashankaryal — APPROVED** [review]
lgtm but i might not have the broadest context in terms of specs

**Review by brimoor — CHANGES_REQUESTED** [review]
Thanks for adding this!

**Functional comment:** can you also handle the case where it is an empty *dataset* rather than a view? When the dataset itself is empty, there's no need for a "clear view" button, the user just needs to know that their dataset is empty:

```py
import fiftyone as fo

dataset = fo.Dataset()
session = fo.launch_app(dataset)
```

**For fun:** what do you think of showing the 404 image from Teams instead? Potential layout:

```
[image]
Your view is empty
[Clear view button]
```

Screen Shot 2023-03-06 at 9 04 15 PM

**Comment by manivoxel51** [comment]
> Thanks for adding this!
> 
> **Functional comment:** can you also handle the case where it is an empty _dataset_ rather than a view? When the dataset itself is empty, there's no need for a "clear view" button, the user just needs to know that their dataset is empty:
> 
> ```python
> import fiftyone as fo
> 
> dataset = fo.Dataset()
> session = fo.launch_app(dataset)
> ```
> 
> **For fun:** what do you think of showing the 404 image from Teams instead? Potential layout:
> 
> ```
> [image]
> Your view is empty
> [Clear view button]
> ```
> 
> Screen Shot 2023-03-06 at 9 04 15 PM

good idea!
I am thinking having both use the same design (our 404 page in teams) make sense too. I'll adapt the pr to
1. show empty page with image + `Your view is empty` title as text and `clear view` as button when there are view stages
2. show empty page with image + `Your dataset is empty` as title, no clear button. also, maybe add a code snippet here to show how to bring data into an existing dataset? or just leave the code snippet out

@brimoor what do you think?

**Comment by brimoor** [comment]
Rather than including a code snippet for the empty dataset case, you could just link to this page on the topic:
https://docs.voxel51.com/user_guide/dataset_creation/index.html

**Review by brimoor — APPROVED** [review]
LGTM

**Pull request merged**